src/routes/RequestsRoute.test.js

```diff
@@ -1,61 +1,144 @@
+      <button type="button" onClick={onBeforeGetContent}>onBeforeGetContent</button>
+      {children}
+    </div>
+  )),
   PrintContent: jest.fn(() => null),
   LoadingButton: jest.fn(() => null),
 }));
+jest.mock('../ViewRequest', () => jest.fn());
+jest.mock('../RequestForm', () => jest.fn());
+
+global.fetch = jest.fn(() => Promise.resolve({
+  json: () => Promise.resolve({ requests: [{ requestLevel: REQUEST_LEVEL_TYPES.TITLE }] }),
+}));
+
 
 const testIds = {
   searchAndSort: 'searchAndSort',
 };
 
+const RequestFilterData = {
+  onChange: jest.fn(),
+};
+
+const request = {
+  requesterId: 'requestId',
+  instanceId: 'instanceId',
+  itemId: 'itemId',
+};
+
+const resultFormatterData = {
+  instance: {
+    publication: [
+      {
+        dateOfPublication: '2023-08-11'
+      }
+    ],
+    title: '',
+  },
+  item: {
+    barcode: 'itemBarcode'
+  },
+  position: 'positon',
+  proxy: {
+    personal: {
+      lastName: 'lastName',
+      firstName: 'firstName',
+      middleName: 'middleName'
+    }
+  },
+  status: '',
+  requester: {
+    barcode: 'requesterBarcode',
+    firstName: 'requesterFirstName',
+    lastName: 'requesterLastName',
+  },
+  requestDate: '2023-04-21',
+  requestType: 'requestType',
+};
+
 SearchAndSort.mockImplementation(jest.fn(({
+  actionMenu,
+  detailProps: { onDuplicate, buildRecordsForHoldsShelfReport, onChangePatron, joinRequest },
+  getHelperResourcePath,
+  massageNewRecord,
+  onCloseNewRecord,
+  onFilterChange,
   parentResources: { records: { records } },
-  detailProps: { onDuplicate },
-}) => (
-  // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
-  <div
-    data-testid={testIds.searchAndSort}
-    onClick={() => onDuplicate(records[0])}
-  />
-)));
-
-jest.mock('../ViewRequest', () => jest.fn());
-jest.mock('../RequestForm', () => jest.fn());
+  renderFilters,
+  resultsFormatter,
+  resultIsSelected,
+  viewRecordOnCollapse
+}) => {
+  const resultsFormatterHandler = () => {
+    resultsFormatter.itemBarcode(resultFormatterData);
+    resultsFormatter.position(resultFormatterData);
+    resultsFormatter.proxy(resultFormatterData);
+    resultsFormatter.requestDate(resultFormatterData);
+    resultsFormatter.requester(resultFormatterData);
+    resultsFormatter.requesterBarcode(resultFormatterData);
+    resultsFormatter.requestStatus(resultFormatterData);
+    resultsFormatter.type(resultFormatterData);
+    resultsFormatter.title(resultFormatterData);
+    resultsFormatter.year(resultFormatterData);
+    resultsFormatter.callNumber(resultFormatterData);
+    resultsFormatter.servicePoint(resultFormatterData);
+  };
+  const onClickActions = () => {
+    onDuplicate(records[0]);
+    buildRecordsForHoldsShelfReport();
+    getHelperResourcePath();
+    massageNewRecord({});
+    resultsFormatterHandler();
+    renderFilters(RequestFilterData.onChange);
+    resultIsSelected({ item: { id: 'id' } });
+    viewRecordOnCollapse();
+  };
```

Comment:
Could you clarify why you need it?
Using this approach you do not test existing functionality instead of that you trigger a lot of methods that help to increase code coverage.

Reply:
resultIsSelected({ item: { id: 'id' } });
viewRecordOnCollapse();
massageNewRecord({});

above 3 methods are used to update the state of object in the code.

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  createRef: jest.fn(),
}));

used above mock to clear console errors